app/helpers/common/priceFormatter.js

```diff
@@ -1,16 +1,22 @@
 export const formatPrice = (priceValue) => {
   const formattedPrice = priceValue.toLocaleString(undefined, {
-    minimumFractionDigits: 3, maximumFractionDigits: 3,
+    minimumFractionDigits: 2, maximumFractionDigits: 4,
   });
 
-  const truncatedTo2dp = formattedPrice.substring(0, formattedPrice.indexOf('.') + 3);
-
-  return truncatedTo2dp;
+  return formattedPrice;
 };
 
 export const formatDecimal = (priceValue) => {
-  if ((priceValue || {}).toString().includes('.') && priceValue.toString().split('.')[1].length < 3) {
-    return parseFloat(priceValue).toFixed(2);
+  if (!priceValue) return '0';
+
+  const priceStringArr = priceValue.toString().split('.');
+
+  if (priceStringArr.length === 1) return priceStringArr[0];
+
+  const fractionalLength = priceStringArr[1].length;
+  if (fractionalLength < 5) {
+    return parseFloat(priceValue).toFixed(fractionalLength < 2 ? 2 : fractionalLength);
   }
-  return priceValue;
+
+  return parseFloat(priceValue).toFixed(4);
 };
```

Comment:
I think we ought to do a bit more refactoring here. I actually think we don't need two functions. There seems to be two main differences between the original versions of `formatPrice` and `formatDecimal`:

1. `formatPrice` returns a string, whereas `formatDecimal` returns a number
2. `formatPrice` always formats to 2 d.p. but `formatDecimal` only formats numbers with a fractional part.

However, in the majority of places `formatDecimal` appears to be used to format numbers for display to users (where returning a string is obviously fine) although there are some places where it appears to be used to parse user input. Other than that the names are confusing. It isn't clear which one a developer should use. Therefore I suggest we should refactor `formatPrice` to this:


```javascript
export const formatPrice = ({ minimumFractionDigits = 2, value }) =>
    parseFloat(value).toLocaleString(undefined, { minimumFractionDigits, maximumFractionDigits: 4 });
```

We then have two options for `formatDecimal` – we refactor it to this:

```javascript
export const formatDecimal = (priceValue) => formatPrice({ minimumFractionDigits: 0, value: priceValue });
```

or we remove it completely.

We will, of course, have to update the usages of `formatPrice` to `formatPrice({ value: whatever })` and if we remove `formatDecimal` we will also to have update its existing usages to `formatPrice({ minimumFractionDigits: 0, value: whatever })`.

I'm in two minds about whether to keep `formatDecimal`, although keeping it means fewer usage updates and could be said to keep things DRY (by reducing `formatPrice({ minimumFractionDigits: 0, value: whatever })` to `formatDecimal(value)`). However, if we do keep it then I think we definitely need to improve naming. Ideally indicate in some way that one always includes decimal places but the other doesn't. And this is why I'm in two minds because I suspect the naming could be awkward and updating `formatDecimal` uses to `formatPrice({ minimumFractionDigits: 0, value: whatever })` makes it absolutely clear that we don't require decimal places in that instance.

Where `formatDecimal` is currently used to parse user input I think we should just be able to replace it with `parseFloat`.

Sorry about the TL;DR. I'll be happy to clarify anything on Monday.

Reply:
After thinking about this a bit more I think something like this might work quite nicely:

```javascript
export const formatNumber = ({ minimumFractionDigits = 0, maximumFractionDigits = 4, value }) =>
    parseFloat(value).toLocaleString(undefined, { minimumFractionDigits, maximumFractionDigits });

export const formatPrice = (value) => formatNumber({ minimumFractionDigits: 2, value });
```